packages/demo/test/Solc.spec.ts

```diff
@@ -119,9 +119,15 @@ describe('Solidity Compiler', () => {
     const errorMessages: Array<string> = []
     results.forEach(({ version, stdout, stderr, code }) => {
       if (code !== 0) {
-        errorMessages.push(
-          `Build failed for ${version} with optimizer enabled.\nSTDOUT: ${stdout}\nSTDERR: ${stderr}`
-        )
+        if (stderr.includes('Unknown version provided')) {
+          console.log(
+            `Detected solc version not currently supported by foundry: ${version}`
+          )
+        } else {
+          errorMessages.push(
+            `Build failed for ${version} with optimizer enabled.\nSTDOUT: ${stdout}\nSTDERR: ${stderr}`
+          )
+        }
```

Comment:
I was running into an error: Unknown version provided in these tests which I do not think is related to my changes. I believe that solc 0.8.24 was recently released and does not appear to be supported by the foundry yet. So I just added logic to handle this specific situation.

Let me know if you disagree.

Reply:
Confirmed that Foundry doesn't support 0.8.24 yet